environment/allocations.go

```diff
@@ -38,15 +38,22 @@ func (a *Allocations) Bindings() nat.PortMap {
 				continue
 			}
 
-			binding := []nat.PortBinding{
-				{
-					HostIP:   ip,
-					HostPort: strconv.Itoa(port),
-				},
+			binding := nat.PortBinding{
+				HostIP:   ip,
+				HostPort: strconv.Itoa(port),
 			}
 
-			out[nat.Port(fmt.Sprintf("%d/tcp", port))] = binding
-			out[nat.Port(fmt.Sprintf("%d/udp", port))] = binding
+			if tcpPort, ok := out[nat.Port(fmt.Sprintf("%d/tcp", port))]; !ok {
```

Comment:
`if v, ok := ...; ok {` would be more in-line stylistically (and logically) with the rest of the code base. I don't like doing the negated check as the primary statement here, it was confusing because in my head the negated should be in the else block.

It reads more like "if there is a value, append it, otherwise create the binding element" rather than what you have which is "if there is not a value create it, otherwise append it." Perhaps that is just my brain though.

Reply:
Agreed, I initially did it that way but changed it because @matthewpi's comment had `!ok`.

How about this:
```go
binding := nat.PortBinding{
	HostIP:   ip,
	HostPort: strconv.Itoa(port),
}

tcpKey := nat.Port(fmt.Sprintf("%d/tcp", port))
udpKey := nat.Port(fmt.Sprintf("%d/udp", port))

if tcpPort, ok := out[tcpKey]; ok {
	out[tcpKey] = append(tcpPort, binding)
} else {
	out[tcpKey] = []nat.PortBinding{binding}
}

if udpPort, ok := out[udpKey]; ok {
	out[udpKey] = append(udpPort, binding)
} else {
	out[udpKey] = []nat.PortBinding{binding}
}
```

Reply:
Yeah flip the `ok`, I just left it how it was because I was ignoring that there was an else case.

Reply:
I prefer the `if v, ok :=` (emphasis on the `v`) since it is more idiomatic and since tcpPort/udpPort are just superfluous since they're scoped only to the if/else block as is.

Reply:
I'd actually go for
```
if _, ok := out[tcpKey]; ok {
  out[tcpKey] = append(out[tcpKey], binding)
} // ...
```
here, since that's much easier to read imho.